### ISSUE TITLE
[enhancement]: Enable Smart Lock C210 (T8502; E110) compatibility in plugin

### ISSUE THREAD

**Issue by codychamberlain** [homebridge-eufy-security/plugin#523]
### Describe the solution you'd like

According to [https://bropat.github.io/eufy-security-client/#/supported_devices] it looks as though Smart Lock C210 (T8502; E110) is fully supported. 

The plugin, however, doesn't seem to have the lock enabled:

> [03/11/2024, 11:46:27]  [EufySecurity]  INFO    Registering new accessory: Patio Door
> [03/11/2024, 11:46:27]  [EufySecurity]  INFO    Registering new accessory: Front Door
> [03/11/2024, 11:46:29]  [EufySecurity]  WARN    This accessory is not compatible with HomeBridge Eufy Security plugin: Patio Door Type: 180
> [03/11/2024, 11:46:29]  [EufySecurity]  WARN    This accessory is not compatible with HomeBridge Eufy Security plugin: Front Door Type: 180
> [03/11/2024, 11:46:29]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemCurrentStateGets: Wrong return value
> [03/11/2024, 11:46:29]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemTargetStateGet: Wrong return value
> [03/11/2024, 11:46:29]  [EufySecurity]  ERROR   Front Door handleSecuritySystemCurrentStateGets: Wrong return value
> [03/11/2024, 11:46:29]  [EufySecurity]  ERROR   Front Door handleSecuritySystemTargetStateGet: Wrong return value

**Comment by codychamberlain** [comment]
[Road Map](https://github.com/homebridge-eufy-security/plugin/projects/1) doesn't seem to be working. Apologies if this is hosted somewhere else and I missed it.

**Comment by slowpc1** [comment]
Looks like the locks were just added this month 3/3 to the device list there at the start of the month. Hopefully it will be updated and added to function correctly in Homekit. (Just got one myself and was wondering about support)

**Comment by Antergosgeek** [comment]
Yes, hoping so, too. I just recently bought a C210 (E110) myself. It currently shows up but can not actually lock or unlock in the HomeKit app.
Hope it’ll work soon

**Comment by lenoxys** [comment]
is the status inversed ? can you described more the expected status ?

**Comment by slowpc1** [comment]
> is the status inversed ? can you described more the expected status ?

Current status is the devices are not supported in the latest build plugin , just the database of devices some of the plugin is based on is not updated to reflect it. 

I believe.

**Comment by lenoxys** [comment]
did you tried on 2.3.11+

**Comment by slowpc1** [comment]
I ran a few betas a few days ago up to latest at that time. Not in front of container so can’t give exact release number but will update today when available.

**Comment by Antergosgeek** [comment]
> is the status inversed ? can you described more the expected status ?

The lock is not inverted it shows correctly if open or closed when I click on at it but says „UNKNOWN“ and shows an open symbol in the HomeKit Dashboard even though when I click on the tile it then shows that it is locked which is also the actual state of the lock. See pictures.
![IMG_8599](https://github.com/homebridge-eufy-security/plugin/assets/26154541/1da39646-a7b7-4832-bb9a-0e434e0f9ccb)
![IMG_8598](https://github.com/homebridge-eufy-security/plugin/assets/26154541/4d7f5659-0483-4d71-b13d-3e7228922cf3)

Hope this helps.

**Comment by Antergosgeek** [comment]
![IMG_8600](https://github.com/homebridge-eufy-security/plugin/assets/26154541/465c3f5e-48d9-433b-af26-b30183fe533f)

**Comment by Antergosgeek** [comment]
Sorry for the multiple Comments, best I can do on  phone.
![IMG_8601](https://github.com/homebridge-eufy-security/plugin/assets/26154541/9acdd5ec-aa7b-4327-a1d2-1467e961f08b)

**Comment by lenoxys** [comment]
`unlock status` should be fixed on 2.3.14. video streaming is still broken.

**Comment by codychamberlain** [comment]
Running version 2.3.14 - the locks are identified but getting this error:
[03/19/2024, 16:48:49]  [EufySecurity]  INFO    Updating existing accessory: Patio Door
[03/19/2024, 16:48:49]  [EufySecurity]  INFO    Updating existing accessory: Front Door
[03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemCurrentStateGets: Wrong return value
[03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemTargetStateGet: Wrong return value
[03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Front Door handleSecuritySystemCurrentStateGets: Wrong return value
[03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Front Door handleSecuritySystemTargetStateGet: Wrong return value

**Comment by Antergosgeek** [comment]
> `unlock status` should be fixed on 2.3.14. video streaming is still broken.

Mine is working perfectly now shows status correctly and everything.

![IMG_8616](https://github.com/homebridge-eufy-security/plugin/assets/26154541/5840c5b2-4e2c-42bc-8c0d-bdf870f8a14a)
![IMG_8617](https://github.com/homebridge-eufy-security/plugin/assets/26154541/bbf63dc4-5173-44a0-a034-7382ebf91258)

**Comment by lenoxys** [comment]
Can you open a new issue for this thanks ?

And provide the full logs ?

> Running version 2.3.14 - the locks are identified and am no longer getting a "not compatible error"  but still getting this error:
> 
> [03/19/2024, 16:48:49]  [EufySecurity]  INFO    Updating existing accessory: Patio Door
> 
> [03/19/2024, 16:48:49]  [EufySecurity]  INFO    Updating existing accessory: Front Door
> 
> [03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemCurrentStateGets: Wrong return value
> 
> [03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Patio Door handleSecuritySystemTargetStateGet: Wrong return value
> 
> [03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Front Door handleSecuritySystemCurrentStateGets: Wrong return value
> 
> [03/19/2024, 16:48:52]  [EufySecurity]  ERROR   Front Door handleSecuritySystemTargetStateGet: Wrong return value
> 
>

**Comment by lenoxys** [comment]
**Thank you for taking the time to report this issue.**

As part of a repository cleanup, we are closing issues that have been open for more than 12 months without resolution. A **beta version** of the plugin has been released that includes significant changes and improvements.

We encourage you to try the latest beta and see if your issue has been addressed. If the problem persists, please feel free to open a new issue with updated details and logs from the latest version.

Thank you for your patience and support!
